src/rp66.cpp

```diff
@@ -0,0 +1,364 @@
+#include <cassert>
+#include <limits>
+#include <vector>
+
+#include <fmt/format.h>
+
+#include <lfp/protocol.hpp>
+#include <lfp/rp66.h>
+
+namespace lfp { namespace {
+
+class rp66 : public lfp_protocol {
+public:
+    rp66(lfp_protocol*);
+
+    // TODO: there must be a "reset" semantic for when there's a read error to
+    // put it back into a valid state
+
+    void close() noexcept (false) override;
+    lfp_status readinto(void* dst, std::int64_t len, std::int64_t* bytes_read)
+        noexcept (false) override;
+
+    int eof() const noexcept (true) override;
+    std::int64_t tell() const noexcept (true) override;
+    void seek(std::int64_t) noexcept (false) override;
+    lfp_protocol* peel() noexcept (false) override;
+    lfp_protocol* peek() const noexcept (false) override;
+
+private:
+    struct header {
+        std::uint16_t  length;
+        char           format;
```

Comment:
Maybe unsigned char or similar?
Debugger thinks it's overflowing and shows me a negative value...